sdk/identity/azure-identity/src/samples/Azure-Cache-For-Redis/Jedis/java/com/azure/jedis/sample/AuthenticateWithTokenCache.java

```diff
@@ -63,12 +66,12 @@ public static void main(String[] args) {
                 // Check if the client is broken, if it is then close and recreate it to create a new healthy connection.
                 if (jedis.isBroken()) {
                     jedis.close();
-                    jedis = createJedisClient(cacheHostname, 6380, "<USERNAME>", tokenRefreshCache.getAccessToken(), useSsl);
+                    accessToken = tokenRefreshCache.getAccessToken();
+                    jedis = createJedisClient(cacheHostname, 6380, extractUsernameFromToken(accessToken.getToken()), accessToken, useSsl);
```

Comment:
use the `username` var from line 41 rather than extracting again? User name won't ever change as long as the same identity is being used

Reply:
updated to reuse username